src/DefinitionResolver.php

```diff
@@ -493,21 +493,37 @@ private function resolveScopedPropertyAccessExpressionNodeToFqn(Node\Expression\
         } elseif ($scoped->scopeResolutionQualifier instanceof Node\QualifiedName) {
             $className = $scoped->scopeResolutionQualifier->getResolvedName();
         }
-        if ($scoped->memberName instanceof Node\Expression\Variable) {
+        $origName = null;
+        do {
+            if ($scoped->memberName instanceof Node\Expression\Variable) {
+                if ($scoped->parent instanceof Node\Expression\CallExpression) {
+                    return null;
+                }
+                $memberName = $scoped->memberName->getName();
+                if (empty($memberName)) {
+                    return null;
+                }
+                $name = (string)$className . '::$' . $memberName;
+            } else {
+                $name = (string)$className . '::' . $scoped->memberName->getText($scoped->getFileContents());
+            }
```

Comment:
There is a lot of logic here that does not change on each loop iteration, such as the `instanceof` checks above and the member name. Could you save this in variables and move it out of the loop?